spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/ReactiveRedisStreamMessageProducer.java

```diff
@@ -0,0 +1,207 @@
+	 * has no effect.
+	 * @param createConsumerGroupIfNotExist specify if we should create the Consumer Group, {@code false} by default
+	 */
+	public void setCreateConsumerGroupIfNotExist(boolean createConsumerGroupIfNotExist) {
+		this.createConsumerGroupIfNotExist = createConsumerGroupIfNotExist;
+	}
+
+	/**
+	 * Set {@link StreamOperations} used to customize the {@link StreamReceiver}.
+	 * It provides a way to set the polling timeout and the serialization context.
+	 * By default the polling timeout is set to infinite and {@link StringRedisSerializer} is used.
+	 * @param streamReceiverOptions the desired receiver options
+	 * */
+	public void setStreamReceiverOptions(@Nullable StreamReceiver.StreamReceiverOptions<String, ?> streamReceiverOptions) {
+		this.streamReceiverOptions = streamReceiverOptions;
+	}
+
+	@Override
+	protected void onInit() {
+		super.onInit();
+		if (this.streamReceiverOptions != null) {
+			this.streamReceiver = StreamReceiver.create(this.reactiveConnectionFactory, this.streamReceiverOptions);
+		}
+		else {
+			this.streamReceiver = StreamReceiver.create(this.reactiveConnectionFactory);
+		}
+		this.evaluationContext = ExpressionUtils.createStandardEvaluationContext(getBeanFactory());
+	}
+
+	@Override
+	protected void doStart() {
+		super.doStart();
+
+		this.streamKey = this.streamKeyExpression.getValue(this.evaluationContext, String.class);
+
+		Assert.notNull(this.streamKey, "'streamKey' must not be null");
+
+		StreamOffset<String> offset = StreamOffset.create(this.streamKey, this.readOffset);
+
+		Flux<Message<?>> messageFlux;
+
+		if (StringUtils.isEmpty(this.consumerGroupName)) {
+			messageFlux = this.streamReceiver
+					.receive(offset)
+					.map(event -> MessageBuilder.withPayload(this.extractPayload ? event.getValue() : event)
+							.setHeader(RedisHeaders.STREAM_KEY, event.getStream())
+							.setHeader(RedisHeaders.STREAM_MESSAGE_ID, event.getId())
+							.build());
+		}
+		else {
+			createConsumerGroup();
+			Consumer consumer = Consumer.from(this.consumerGroupName, this.consumerName);
+			messageFlux = this.streamReceiver
+					.receiveAutoAck(consumer, offset)
+					.map(event -> MessageBuilder.withPayload(this.extractPayload ? event.getValue() : event)
+							.setHeader(RedisHeaders.STREAM_KEY, event.getStream())
+							.setHeader(RedisHeaders.STREAM_MESSAGE_ID, event.getId())
+							.build());
+		}
+
+		subscribeToPublisher(messageFlux);
+	}
+
+	@Override
+	public String getComponentType() {
+		return "redis:stream-inbound-channel-adapter";
+	}
+
+	private void createConsumerGroup() {
+		if (this.createConsumerGroupIfNotExist) {
+			Assert.hasText(this.consumerGroupName, "'consumerGroupName' must be set");
+			Assert.hasText(this.consumerName, "'consumerName' must be set");
+
+			this.reactiveConnectionFactory.getReactiveConnection()
+					.streamCommands()
+					.xGroupCreate(ByteBuffer.wrap(this.streamKey.getBytes(StandardCharsets.UTF_8)),
```

Comment:
Let's make this as a part of the `ReactiveStreamOperations.createGroup()` reactive stream!

I mean it is better to do such a `Mono` call in that `doStart()` and then do `thenMany()`  for that `this.streamReceiver.receive()`.